src/core/MOM_dynamics_split_RK2b.F90

```diff
@@ -538,14 +542,15 @@ subroutine step_MOM_dyn_split_RK2b(u_av, v_av, h, tv, visc, Time_local, dt, forc
     call open_boundary_zero_normal_flow(CS%OBC, G, GV, CS%PFu, CS%PFv)
 
   if (G%nonblocking_updates) then
-    call complete_group_pass(CS%pass_hp_uhvh, G%Domain, clock=id_clock_pass)
+    call complete_group_pass(CS%pass_hp_uv, G%Domain, clock=id_clock_pass)
     call start_group_pass(CS%pass_eta, G%Domain, clock=id_clock_pass)
   else
-    call do_group_pass(CS%pass_hp_uhvh, G%Domain, clock=id_clock_pass)
+    call do_group_pass(CS%pass_hp_uv, G%Domain, clock=id_clock_pass)
   endif
 
+  call do_group_pass(CS%pass_h, G%Domain, clock=id_clock_pass)
```

Comment:
There were calls in MOM.F90 that already updated a 2-point wide halo for h, so this new `do_group_pass()` call is only needed if `cor_stencil > 2`.  Also, it would be more efficient if it were conditionally added to the same group pass with hp, uh, vh, u and v.